viewer/packages/cad-model/src/utilities/TreeIndexToSectorsMap.ts

```diff
@@ -11,24 +11,28 @@ import { RevealGeometryCollectionType } from '@reveal/sector-parser';
  */
 export class TreeIndexToSectorsMap {
   public onChange?: (treeIndex: number, newSectorId: number) => void;
-  private readonly _treeIndexToSectorIds = new Map<number, Set<number>>();
+  private readonly _treeIndexToSectorIds: (number[] | undefined)[];
   private readonly _parsedSectors = new Map<number, Set<RevealGeometryCollectionType>>();
 
+  constructor(maxTreeIndex: number) {
+    this._treeIndexToSectorIds = Array(maxTreeIndex);
+  }
+
   /**
    * Store the fact that a tree index is found to have geometry in a certain sector
    * @param treeIndex Tree index
    * @param sectorId The sector id where the tree index was found
    */
   set(treeIndex: number, sectorId: number): void {
-    const existingSet = this._treeIndexToSectorIds.get(treeIndex);
+    const existingSet = this._treeIndexToSectorIds[treeIndex];
```

Comment:
Might consider renaming `existingSet` to something less error-prone (speaking from recent experience.. :stuck_out_tongue: ). `existingSectorIds` or just `sectorIds` or something along those lines.